File: ruby/resistor-color-duo/resistor_color_duo.rb
# frozen_string_literal: true

#:nodoc:
class ResistorColorDuo
  COLOUR_VALUES = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8
  }.freeze
  def self.value(colours)
    colours.first(2).reverse.each_with_index.inject(
      0
    ) do |value, (colour, index)|
      value + (10**index * COLOUR_VALUES[colour.to_sym])
    end
  end
end

File: typescript/resistor-color-duo/resistor-color-duo.ts
export const COLORS: string[] = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
]

export function decodedValue(bands: string[]) {
  return parseInt(
    bands
      .splice(0, 2)
      .map((band) => COLORS.indexOf(band))
      .join("")
  )
}
